src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/ConnectHelper.cs

```diff
@@ -54,7 +86,7 @@ public static async ValueTask<(Socket, Stream)> ConnectAsync(string host, int po
                 saea.Initialize(cancellationToken);
 
                 // Configure which server to which to connect.
-                saea.RemoteEndPoint = new DnsEndPoint(host, port);
+                saea.RemoteEndPoint = new DnsEndPoint(host, port, family);
```

Comment:
It looks like this change could end up causing Socket to go down a path that hits a similar issue to the one called out in https://github.com/dotnet/corefx/issues/29285.